samples/java-spring-tracing/src/main/java/com/example/tracing/RestTemplateConfig.java

```diff
@@ -0,0 +1,22 @@
+package com.example.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.spring.web.v3_1.SpringWebTelemetry;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+public class RestTemplateConfig {
+
+  // this was added just for testing the auto-instrumentation
+  @Bean
+  public RestTemplate restTemplate(OpenTelemetry openTelemetry) {
+
+    RestTemplate restTemplate = new RestTemplate();
+    SpringWebTelemetry telemetry = SpringWebTelemetry.create(openTelemetry);
+    restTemplate.getInterceptors().add(telemetry.newInterceptor());
```

Comment:
I understand that this is intercepting the call to the endpoint and injects the `traceparent` with the span info from the `OpenTelemetry` context in `ThreadLocal`. In think this is the usual way anybody would use tracing. Relying on the `OpenTelemetry.currentContext()`